### ISSUE TITLE
Table Panel: Cell popover menu

### ISSUE THREAD

**Issue by gtk-grafana** [grafana/grafana#81900]
TL;DR Calling component of `<Table>` can provide a "popover" menu that is shown on hover of a table cell.

**Why is this needed**:
Table panel is used for visualizing data throughout grafana applications, however besides visualization only filtering/sorting, and filter for/out value the table does not allow much interaction.

**What would you like to be added**:
A callback function passed into the `<Table>` component props that is triggered when the user selects text within a table cell, or hovers over a table cell.

The function would need to return the selected string, a unique identifier for the field (like name or id), and the index of the field value.
The calling component would also need to pass in some react component that the table will need to visualize after the user has selected text within a table cell.

Mockup in table viz: 
<img width="252" alt="image" src="https://github.com/grafana/grafana/assets/109082771/0e6b6030-5791-4517-afc7-e0a9183b27ef">

Examples in logs explore visualization: 
<img width="492" alt="image" src="https://github.com/grafana/grafana/assets/109082771/0914413e-9e8b-453d-822f-633938930e22">

<img width="200" alt="image" src="https://github.com/grafana/grafana/assets/109082771/68df805c-8b20-4b6c-88cd-83f8824aec24">

**Who is this feature for?**
Developers building on top of table component: This would allow o11y-logs to bring the popover menu into the existing table view in explore

**Comment by matyax** [comment]
From my experience with LogRow:

> A callback function passed into the <Table> component props that is triggered when the user selects text within a table cell, or hovers over a table cell.

We use `onClick` to decide to show the menu or not: https://github.com/grafana/grafana/blob/main/public/app/features/logs/components/LogRow.tsx#L106 . 

A callback for other events like hover, mouse enter, mouse leave, can potentially be useful for a log row menu or something similar.

**Comment by gtk-grafana** [comment]
For a solution in the HoC see this PR: https://github.com/grafana/grafana/pull/82440#issuecomment-1944092490
